theseus/core/theseus_function.py

```diff
@@ -80,6 +88,13 @@ def copy(
                 new_var.name = old_var.name
             for old_aux, new_aux in zip(self.aux_vars, new_fn.aux_vars):
                 new_aux.name = old_aux.name
+
+        if self._has_duplicate_vars(new_fn):
+            raise RuntimeError(
```

Comment:
Is the desired behavior to prevent reuse altogether or should we raise a warning instead and provide some other default behavior.

Could the error message be a bit cryptic, if it get's triggered from some leaf cost function?